### PR TITLE
asset-swapper: Exchange Proxy support

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

This adds a new swap quote consumer to `asset-swapper`, `ExchangeProxySwapQuoteConsumer`, which can generate call data for the Exchange Proxy's `transformERC20()` function.

`ExchangeProxySwapQuoteConsumer.getCallDataOrThrowAsync()` can accept an options object:
```ts
{
  extensionContractOpts: {
    // Whether to accept ETH and automatically wrap it to WETH prior to fill.
    isFromETH: boolean;
    // Whether to unwrap WETH to ETH after the fill.
    isToETH: boolean;
  }
}
```

The `CalldataInfo` struct returned by `getCallDataOrThrowAsync()` from all consumers now has an `allowanceTarget` field.

### Odds and Ends
- Pinned `@0x/instant`'s `@0x/asset-swapper` dependency to the old version to avoid constantly breaking instant.
- Added transformer data decoding utils to `@0x/contracts-zero-ex`.
- Add `ITransformERC20` contract artifacts/wrappers where necessary.
- Added placeholder EP addresses to `ContractAddresses`. None of this stuff is deployed so they're all NULL for now.
- Fix `@0x/contracts-broker` tests and add them to CI.
- Allow for identifiers prefixed with underscores in python contract wrappers.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/instant/package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "homepage": "https://github.com/0xProject/0x-monorepo/packages/instant/README.md",
     "dependencies": {
         "@0x/assert": "^3.0.7",
-        "@0x/asset-swapper": "^4.4.0",
+        "@0x/asset-swapper": "4.4.0",
```
Comment (dorothy-zbornak) [inline packages/instant/package.json:47]: :safety_pin:

File: packages/contract-wrappers/CHANGELOG.json [hunk 2]
```diff
@@ -21,6 +21,10 @@
             {
                 "note": "Added `ERC20BridgeSampler.sampleBuysFromKyberNetwork`",
                 "pr": 2551
+            },
+            {
+                "note": "Add `ITransformERC20`",
+                "pr": "TODO"
```
Comment (dekz) [inline packages/contract-wrappers/CHANGELOG.json:27]:
```suggestion
                "pr": 2591
```

File: packages/migrations/CHANGELOG.json [hunk 3]
```diff
@@ -9,6 +9,10 @@
             {
                 "note": "Add `ERC20BridgeSampler` deployment",
                 "pr": 2541
+            },
+            {
+                "note": "Return emtpy Exchange Proxy addresses",
```
Comment (dekz) [inline packages/migrations/CHANGELOG.json:14]:
```suggestion
                "note": "Return empty Exchange Proxy addresses",
```

File: packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                buyToken,
+                quote.worstCaseQuoteInfo.totalTakerAssetAmount,
+                quote.worstCaseQuoteInfo.makerAssetAmount,
+                transforms,
+            )
+            .getABIEncodedTransactionData();
+
+        return {
+            calldataHexString,
+            ethAmount: quote.worstCaseQuoteInfo.protocolFeeInWeiAmount,
```
[the remark this answers is not in the document]
Comment (dorothy-zbornak) [reply packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts:130]: lol doh

File: packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+        // Build up the transforms.
+        const transforms = [];
+        if (exchangeProxyOpts.isFromETH) {
+            // Create a WETH wrapper if coming from ETH.
+            transforms.push({
+                transformer: this.contractAddresses.transformers.wethTransformer,
+                data: encodeWethTransformerData({
+                    token: ETH_TOKEN_ADDRESS,
+                    amount: quote.worstCaseQuoteInfo.totalTakerAssetAmount,
```
[the remark this answers is not in the document]
Comment (dorothy-zbornak) [reply packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts:79]: No, because we don't want to wrap the protocol fee.

**Pull request merged**